### PR TITLE
Migrate ADO Samples from ADAL to MSAL

### PR DESCRIPTION
**Pull request by Shama-K**
This PR is created to revise [ManagedClientConsoleAppSample](https://github.com/microsoft/azure-devops-auth-samples/tree/master/ManagedClientConsoleAppSample) by replacing ADAL.NET with MSAL.NET
Count for https://github.com/microsoft/azure-devops-auth-samples/issues/56.
Edit: Update other .NET samples from the repo.

cc: @kalyankrishna1

### CODE REVIEW

File: ManagedClientConsoleAppSample/Program.cs [hunk 1]
```diff
@@ -1,68 +1,81 @@
...
+        internal static string[] scopes = new string[] { "499b84ac-1321-427f-aa17-267ca6975798/user_impersonation" }; //Constant value to target Azure DevOps. Do not change  
+        
+        // MSAL Public client app
+        private static IPublicClientApplication application;
 
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            AuthenticationContext ctx = GetAuthenticationContext(null);
+            var accessToken = await SignInUserAndGetTokenUsingMSAL(scopes);
```
Comment (jmprieur) [inline ManagedClientConsoleAppSample/Program.cs:35]: You might want to return the AuthenticationResult, and then use `authenticationResult.CreateAuthorizationHeader` (which works with PoP as well bearer)

**Review by jmprieur — APPROVED** [review]
LGTM.
Thanks @Shama-K

File: ManagedClientConsoleAppSample/Program.cs [hunk 2]
```diff
@@ -1,68 +1,81 @@
...
+        /// <param name="scopes"></param>
+        /// <returns>Access Token</returns>
+        private static async Task<string> SignInUserAndGetTokenUsingMSAL(string[] scopes)
+        {
+            // Initialize the MSAL library by building a public client application
+            application = PublicClientApplicationBuilder.Create(clientId)
+                                       .WithAuthority(authority)
+                                       .WithDefaultRedirectUri()
+                                       .Build();
+
```
Comment (jmprieur) [inline ManagedClientConsoleAppSample/Program.cs:56]: You might want to add token cache serialization.:
https://github.com/AzureAD/microsoft-identity-web/blob/b15e6a216805f4bee77894915b0f06599d0651e4/tools/app-provisioning-tool/app-provisioning-lib/DeveloperCredentials/MsalTokenCredential.cs#L49

and 

https://github.com/AzureAD/microsoft-identity-web/blob/b15e6a216805f4bee77894915b0f06599d0651e4/tools/app-provisioning-tool/app-provisioning-lib/DeveloperCredentials/MsalTokenCredential.cs#L71-L72

**Comment by jmprieur** [comment]
@aiwangmicrosoft

File: ManagedClientConsoleAppSample/Program.cs [hunk 3]
```diff
@@ -1,69 +1,91 @@
...
+            try
             {
-                ctx = new AuthenticationContext("https://login.windows.net/common");
-                if (ctx.TokenCache.Count > 0)
-                {
-                    string homeTenant = ctx.TokenCache.ReadItems().First().TenantId;
-                    ctx = new AuthenticationContext("https://login.microsoftonline.com/" + homeTenant);
-                }
+                var accounts = application.GetAccountsAsync().Result;
```
Comment (jmprieur) [inline ManagedClientConsoleAppSample/Program.cs:70]: why not:

```CSharp
var accounts = await application.GetAccountsAsync();
```

File: DeviceProfileSample/README.md [hunk 4]
```diff
@@ -1,25 +1,27 @@
 # Device Profile Sample
 
-For a headless text output client application, it is not possible authenticate through an interactive prompt. Instead a text only approach is necessary. This flow leverages a user's external device (i.e. phone) to authenticate through an interactive login prompt and pass the auth token to the headless application. For more information [click here](https://azure.microsoft.com/en-us/resources/samples/active-directory-dotnet-deviceprofile/?v=17.23h).
+For a headless text output client application, it is not possible authenticate through an interactive prompt. Instead a text only approach is necessary. This flow leverages a user's external device (i.e. phone) to authenticate through an interactive login prompt and pass the auth token to the headless application. For more information [click here](https://azure.microsoft.com/resources/samples/active-directory-dotnet-deviceprofile/?v=17.23h).
```
Comment (jmprieur) [inline DeviceProfileSample/README.md:3]: Maybe say that if the tenant admin requires device authentication conditional access policies, using the Device profile flow won't be a good option ?

File: ManagedClientConsoleAppSample/Program.cs [hunk 5]
```diff
@@ -1,69 +1,91 @@
...
-            if (tenant != null)
-                ctx = new AuthenticationContext("https://login.microsoftonline.com/" + tenant);
-            else
+            // Initialize the MSAL library by building a public client application
+            application = PublicClientApplicationBuilder.Create(clientId)
+                                       .WithAuthority(authority)
+                                       .WithDefaultRedirectUri()
+                                       .Build();
+
+            AuthenticationResult result = null;
```
Comment (jmprieur) [inline ManagedClientConsoleAppSample/Program.cs:66]: = null is not needed, I'd think

**Review by jmprieur — APPROVED** [review]
LGTM. 
Thanks @Shama-K

File: DeviceProfileSample/README.md (hunk 4 above)
Comment (Shama-K) [reply DeviceProfileSample/README.md:3]: Added the info in Readme.
Thank you.

**Review by kalyankrishna1 — APPROVED** [review]

**Review by anydot — APPROVED** [review]

**Pull request merged**